src/mjolnir/pbfgraphparser.cc

```diff
@@ -2294,18 +2291,22 @@ struct graph_callback : public OSMPBF::Callback {
     osmdata_.edge_count -= intersection; // more accurate but undercounts by skipping lone edges
   }
 
-  virtual void way_callback(const uint64_t osmid,
-                            const OSMPBF::Tags& tags,
-                            const std::vector<uint64_t>& nodes) override {
-
-    osmid_ = osmid;
+  void way(const osmium::Way& way) {
+    changeset(way.changeset());
 
+    osmid_ = static_cast<uint64_t>(way.id());
     // unsorted extracts are just plain nasty, so they can bugger off!
     if (osmid_ < last_way_) {
       throw std::runtime_error("Detected unsorted input data");
     }
     last_way_ = osmid_;
 
+    std::vector<uint64_t> nodes;
```

Comment:
would it be beneficial to have this sitting outside of the function, long-lived somewhere, so that we can reuse it (clearing on each run). to avoid constant de/reallocs

Reply:
I like that idea, but I doubt that this optimisation can be noticed among current amount of allocations per way processed.

Reply:
I'm just thinking that this part would be changed soon anyway during even first steps towards multithreaded parsing.